index/secondary_index.cc

```diff
@@ -106,6 +106,26 @@ sstring target_parser::get_target_column_name_from_string(const sstring& targets
     return targets;
 }
 
+sstring target_parser::get_base_column_name_from_target(const sstring& target) {
+    sstring base_name = target;
+    
+    // extract column name if the target is in format "function(column_name)"
+    auto opening_pos = target.find("(");
+    if (opening_pos != target.npos) {
+        auto closing_pos = target.find(")", ++opening_pos);
+        if (opening_pos != target.npos) {
+            base_name = target.substr(opening_pos, closing_pos - opening_pos);
```

Comment:
This operation would also be more efficient if `base_name`/`target` were `std::string_views`.